src/simulation/m_global_parameters.fpp

```diff
@@ -1068,6 +1068,11 @@ contains
             buff_size = weno_polyn + 2
         end if
 
+        if (probe_wrt) then
+            fd_number = max(1, fd_order/2)
+            buff_size = buff_size + fd_number
```

Comment:
That's right, only the fd_number needs to be updated for using a probe. MFC 4.9.6 has the same issue (buff_size depends on whether or not using a probe) because the halo region was alternated several times in the `s_compute_rhs()`. Fixed.